libs/ledger-live-common/src/families/cosmos/js-synchronisation.ts

```diff
@@ -54,130 +55,105 @@ const txToOps = (info: AccountShapeInfo, accountId: string, txs: CosmosTx[]): Op
-          const amountAttributes = message.attributes.filter(
-            attribute =>
-              attribute.key === "amount" && attribute.value.includes(currency.units[1].code),
-          );
-
-          amountAttributes.forEach(amountAttribute => {
-            messageRewardValue = messageRewardValue.plus(
-              new BigNumber(amountAttribute.value.replace(currency.units[1].code, "")),
-            );
-          });
-
-          rewardShards.push({
-            amount: messageRewardValue,
-            address: validatorAddress,
-          });
-
-          txRewardValue = txRewardValue.plus(messageRewardValue);
         }
-
         op.value = txRewardValue;
         op.extra.validators = rewardShards;
-
         break;
       }
-
-      case "delegate":
-        if (attributes.amount && attributes.amount.indexOf(currency.units[1].code) != -1) {
-          op.type = "DELEGATE";
-          op.value = new BigNumber(fees);
-          op.extra.validators.push({
-            address: attributes.validator,
-            amount: attributes.amount.replace(currency.units[1].code, ""),
-          });
+      case "delegate": {
+        op.type = "DELEGATE";
+        op.value = new BigNumber(fees);
+        const delegateShards: { amount: BigNumber; address: string }[] = [];
+        for (const message of correspondingMessages) {
+          const amount = message.attributes.find(attr => attr.key === "amount")?.value;
+          const validator = message.attributes.find(attr => attr.key === "validator")?.value;
+          if (amount && validator && amount.endsWith(unitCode)) {
+            delegateShards.push({
+              amount: new BigNumber(amount.replace(unitCode, "")),
+              address: validator,
+            });
+          }
         }
+        op.extra.validators = delegateShards;
         break;
-
-      case "redelegate":
-        if (
-          attributes.amount &&
-          attributes.amount.indexOf(currency.units[1].code) != -1 &&
-          attributes.destination_validator &&
-          attributes.source_validator
-        ) {
-          op.type = "REDELEGATE";
-          op.value = new BigNumber(fees);
-          op.extra.validators.push({
-            address: attributes.destination_validator,
-            amount: attributes.amount.replace(currency.units[1].code, ""),
-          });
-          op.extra.sourceValidator = attributes.source_validator;
+      }
+      case "redelegate": {
+        op.type = "REDELEGATE";
+        op.value = new BigNumber(fees);
+        const redelegateShards: { amount: BigNumber; address: string }[] = [];
+        for (const message of correspondingMessages) {
+          const amount = message.attributes.find(attr => attr.key === "amount")?.value;
+          const validatorDst = message.attributes.find(attr => attr.key === "validator_dst")?.value;
+          const validatorSrc = message.attributes.find(attr => attr.key === "validator_src")?.value;
```

Comment:
@hzheng-ledger This introduced a regression: https://ledgerhq.atlassian.net/browse/LIVE-8787

Indeed the attribute keys were changed from `source_validator` and `destination_validator` to `validator_src` and `validator_dst`, was this intentional?

I'm seeing that at least for ATOM the messages still have this structure:
```
[
  {
    key: "source_validator",
    value: "cosmosvaloper1grgelyng2v6v3t8z87wu3sxgt9m5s03xfytvz7",
  },
  {
    key: "destination_validator",
    value: "cosmosvaloper10wljxpl03053h9690apmyeakly3ylhejrucvtm",
  },
  {
    key: "amount",
    value: "5031uatom",
  },
  {
    key: "completion_time",
    value: "2022-12-07T16:41:08Z",
  },
]
```